### PR TITLE
makes the web weaving mutation let you weave webs into cloth

### PR DESCRIPTION
**Pull request by Clyde-Thunderpants**
## About The Pull Request
essentially, if you have the mutation that allows you to lay loads of webs to annoy people, you can click a web to weave it into cloth. Craft bandages, damp rags, duffelbags, or whatever else cloth is used for to your heart's content, instead of just using it to annoy sec. This is part of the ongoing effort by @KathyRyals to make more genetics powers have actual utility. 
I was going to do a cool radial menu where you got to pick what to make, maybe add some silk sutures or something for medbay as a shitty improvised way of healing, but then I couldn't be bothered because I don't know what a radial menu is so I just did this instead.
There's a new trait, TRAIT_WEB_WEAVER which gives this behaviour, and injecting the webbing mutation gives you it. Admins could also give you the trait I suppose.

## Why It's Good For The Game
gives marginal utility to a power basically used for griefing the sec hallway and nothing else. Even if you don't actually need the cloth, it's a good way of clearing out a lot of cobwebs in a hurry if you've got no lighter.

## Changelog
:cl:
add: New trait, web weaver, that allow you to weave webs into cloth.
add: The genetics webbing mutation now gives you web weaver
/:cl:

### CODE REVIEW

**Comment by ilikesaying** [comment]
>moths now have an infinite supply of food

**Comment by RaveRadbury** [comment]
@Clyde-Thunderpants if you want to try doing radials I can show you some PR's that I did, also @Arkatos1 has done a lot of radial menu work, his PRs are great reads.

**Comment by Nexofas** [comment]
"gives marginal utility to a power basically used for griefing the sec hallway and nothing else. "
Yes. Giving marginal utility instead of griefing power is what genetics should be in my opinion.

**Comment by moocowswag** [comment]
Ive played around with webs a bit and webs themselves are useful, but admittedly niche and the mutation feels unfinished due to lack of chromosome support. (At the very least it should allow the use of energetic)

Giving them utility / an easy way to produce station resources is pretty big and might even require an instability increase.

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -215,6 +215,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_DISK_VERIFIER "disk-verifier"
 #define TRAIT_NOMOBSWAP "no-mob-swap"
 #define TRAIT_XRAY_VISION "xray_vision"
+#define TRAIT_WEB_WEAVER "web_weaver"
```
Comment (Mothblocks) [inline code/__DEFINES/traits.dm:218]: Document.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/__DEFINES/traits.dm (hunk 1 above)
Comment (Clyde-Thunderpants) [reply code/__DEFINES/traits.dm:219]: should be done, if I correctly understood what you meant by "document"

**Pull request merged**